source/plugins/ruby/filter_cadvisor2mdm.rb

```diff
@@ -308,8 +308,16 @@ def ensure_cpu_memory_capacity_set
         end
       elsif controller_type.downcase == "daemonset"
         capacity_from_kubelet = KubeletUtils.get_node_capacity
-        @cpu_capacity = capacity_from_kubelet[0]
-        @memory_capacity = capacity_from_kubelet[1]
+
+        # Error handling in case /metrics/cadvsior endpoint fails
+        if !capacity_from_kubelet.nil? && capacity_from_kubelet.length > 1
+          @cpu_capacity = capacity_from_kubelet[0]
+          @memory_capacity = capacity_from_kubelet[1]
+        else
+          # cpu_capacity and memory_capacity keep initialized value of 0.0
+          @log.info "Error getting capacity_from_kubelet: cpu_capacity and memory_capacity"
```

Comment:
should be the error rather than info and we should also emit to AI telemetry so that it will be helpful to handle the customer incidents.

Reply:
I will change it to @log.error. I believe AI telemetry will be sent here when KubeletUtils.get_node_capacity fails: https://github.com/microsoft/Docker-Provider/blob/ef9d726c7053fba0254fc897aff124e5a5a2be34/source/plugins/ruby/kubelet_utils.rb#L33

Reply:
> I will change it to @log.error. I believe AI telemetry will be sent here when KubeletUtils.get_node_capacity fails:
> 
> https://github.com/microsoft/Docker-Provider/blob/ef9d726c7053fba0254fc897aff124e5a5a2be34/source/plugins/ruby/kubelet_utils.rb#L33

> Ok, then pls change either error or warn instead of info here. I see many places in existing code, we use info instead of error/warn for the error logs.